app/controllers/AccountManagement.scala

```diff
@@ -134,6 +134,21 @@ object ManageWeekly extends LazyLogging {
 
   import play.api.mvc.Results._
 
+  // this sequencing concatenates errors if any, otherwise aggregates rights
+  def sequence[A](list: List[Either[String, A]]): Either[String, List[A]] = {
+    val errors = list collect {
+      case Left(x) => x
+    }
+    if (errors.nonEmpty)
+      Left(errors.mkString(", "))
+    else
+      Right(list collect {
+        case Right(x) => x
+      })
```

Comment:
What about doing something like this? 

list.partition(_.isLeft) match {
  case (Nil, successResponses) => res.map { case Right(success) => success }
  case (errors, _) => errors.mkString(", ")
}